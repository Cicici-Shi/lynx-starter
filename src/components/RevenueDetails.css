/* Copyright 2024 The Lynx Authors. All rights reserved. */
/* Licensed under the Apache License Version 2.0 that can be found in the */
/* LICENSE file in the root directory of this source tree. */

/* 收益分配详情组件样式 */
.revenue-details {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 1000;
  opacity: 0;
  visibility: hidden;
  transition: opacity 0.3s ease-in-out, visibility 0.3s ease-in-out;
}

.revenue-details.visible {
  opacity: 1;
  visibility: visible;
}

/* 遮罩层 */
.details-overlay {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.5);
}

/* 详情面板 */
.details-panel {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 90%;
  max-width: 600px;
  max-height: 80vh;
  background: var(--rp-panel, #fcfcfc);
  border-radius: 16px;
  box-shadow: var(--rp-shadow, 0 20px 25px -5px rgba(0, 0, 0, 0.1), 0 10px 10px -5px rgba(0, 0, 0, 0.04));
  overflow: hidden;
  animation: slideIn 0.3s ease-out;
}

@keyframes slideIn {
  from {
    opacity: 0;
    transform: translate(-50%, -50%) scale(0.95);
  }
  to {
    opacity: 1;
    transform: translate(-50%, -50%) scale(1);
  }
}

/* 头部 */
.details-header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 20px 24px;
  border-bottom: 1px solid var(--rp-border, #e5e7eb);
  background: var(--rp-bg, #f8f7f6);
}

.details-title {
  font-size: 18px;
  font-weight: 600;
  color: var(--rp-text, #0f172a);
  margin: 0;
}

.header-actions {
  display: flex;
  align-items: center;
  gap: 16px;
}

/* 关闭按钮 */
.close-button {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 32px;
  height: 32px;
  border-radius: 8px;
  cursor: pointer;
  transition: background-color 0.15s ease;
}

.close-button:hover {
  background: var(--rp-border, #e5e7eb);
}

.close-icon {
  color: var(--rp-muted, #6b7280);
}

/* 内容区域 */
.details-content {
  max-height: calc(80vh - 80px);
  overflow-y: auto;
}

/* 基本信息 */
.basic-info {
  padding: 20px 24px;
  border-bottom: 1px solid var(--rp-border, #f3f4f6);
}

.info-item {
  display: flex;
  align-items: flex-start;
  margin-bottom: 12px;
}

.info-item:last-child {
  margin-bottom: 0;
}

.info-label {
  font-size: 14px;
  color: var(--rp-muted, #6b7280);
  font-weight: 500;
  min-width: 80px;
  margin-right: 12px;
}

.info-value {
  font-size: 14px;
  color: var(--rp-text, #0f172a);
  flex: 1;
}

.total-amount {
  font-weight: 600;
  color: var(--rp-brand, #10b981);
  font-size: 16px;
}

.calculation-basis {
  align-items: flex-start;
}

.basis-text {
  line-height: 1.5;
}

/* 分配详情区域 */
.distribution-section {
  padding: 20px 24px;
  border-bottom: 1px solid var(--rp-border, #f3f4f6);
}

.section-title {
  font-size: 16px;
  font-weight: 600;
  color: var(--rp-text, #0f172a);
  margin-bottom: 16px;
  display: block;
}

.distribution-list {
  display: flex;
  flex-direction: column;
  gap: 16px;
}

.distribution-item {
  padding: 16px;
  background: var(--rp-bg, #f8f7f6);
  border-radius: 12px;
  border: 1px solid var(--rp-border, #f3f4f6);
}

.item-header {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  margin-bottom: 8px;
}

.party-info {
  display: flex;
  align-items: center;
  gap: 8px;
  flex: 1;
}

.category-indicator {
  width: 12px;
  height: 12px;
  border-radius: 50%;
  flex-shrink: 0;
}

.party-name {
  font-size: 14px;
  font-weight: 600;
  color: var(--rp-text, #0f172a);
}

.party-category {
  font-size: 12px;
  color: var(--rp-muted, #6b7280);
}

.party-amount {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  gap: 2px;
}

.percentage {
  font-size: 16px;
  font-weight: 700;
  color: var(--rp-text, #0f172a);
}

.amount {
  font-size: 12px;
  color: var(--rp-muted, #6b7280);
}

.distribution-reason {
  font-size: 13px;
  color: var(--rp-muted, #6b7280);
  line-height: 1.4;
  margin-bottom: 12px;
}

/* 百分比可视化条 */
.percentage-bar {
  height: 4px;
  background: var(--rp-border, #e5e7eb);
  border-radius: 2px;
  overflow: hidden;
}

.percentage-fill {
  height: 100%;
  border-radius: 2px;
  transition: width 0.3s ease;
  animation: barGrow 0.6s ease-out;
}

@keyframes barGrow {
  from {
    width: 0%;
  }
}

/* 分配汇总 */
.distribution-summary {
  padding: 20px 24px;
  background: var(--rp-bg, #f8f7f6);
  display: flex;
  justify-content: space-between;
  gap: 20px;
}

.summary-item {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 4px;
}

.summary-label {
  font-size: 12px;
  color: var(--rp-muted, #6b7280);
  font-weight: 500;
}

.summary-value {
  font-size: 16px;
  font-weight: 700;
  color: var(--rp-text, #0f172a);
}

/* 响应式设计 */
@media (max-width: 640px) {
  .details-panel {
    width: 95%;
    max-height: 90vh;
  }

  .details-header {
    padding: 16px 20px;
  }

  .details-title {
    font-size: 16px;
  }

  .basic-info,
  .distribution-section {
    padding: 16px 20px;
  }

  .distribution-summary {
    padding: 16px 20px;
    flex-direction: column;
    gap: 12px;
  }

  .summary-item {
    flex-direction: row;
    justify-content: space-between;
  }
}

/* 动画优化 */
@media (prefers-reduced-motion: reduce) {
  .revenue-details,
  .details-panel,
  .close-button,
  .percentage-fill {
    transition: none;
    animation: none;
  }
}