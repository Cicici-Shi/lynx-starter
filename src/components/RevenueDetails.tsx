// Copyright 2024 The Lynx Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.

import { Icon } from "./Icon.tsx";
import type { RevenueDistribution } from "../types/chatbot.ts";
import type { RevenueDetailsProps } from "../types/components.ts";
import "./RevenueDetails.css";

/**
 * 获取参与方类别的显示颜色
 */
function getCategoryColor(category: string, index: number): string {
  const colors = {
    streamer: "#10b981", // 绿色
    platform: "#3b82f6", // 蓝色
    partner: "#f59e0b", // 橙色
    other: "#6b7280", // 灰色
  };

  // 如果类别不存在，使用备用颜色数组
  const fallbackColors = [
    "#10b981",
    "#3b82f6",
    "#f59e0b",
    "#ef4444",
    "#8b5cf6",
    "#06b6d4",
  ];

  return (
    colors[category as keyof typeof colors] ||
    fallbackColors[index % fallbackColors.length]
  );
}

/**
 * 收益分配详情组件
 *
 * 功能特性:
 * - 展示分配详情和分配比例、金额和理由
 * - 实现平滑的展开/收起动画效果
 * - 显示分配比例可视化条
 */
export function RevenueDetails({
  distribution,
  isVisible,
  onClose,
}: RevenueDetailsProps) {
  /**
   * 格式化金额显示
   */
  const formatAmount = (amount: number): string => {
    return amount.toLocaleString("zh-CN", {
      minimumFractionDigits: 2,
      maximumFractionDigits: 2,
    });
  };

  /**
   * 格式化计算时间
   */
  const formatCalculatedTime = (dateString: string): string => {
    const date = new Date(dateString);
    return date.toLocaleString("zh-CN", {
      year: "numeric",
      month: "2-digit",
      day: "2-digit",
      hour: "2-digit",
      minute: "2-digit",
    });
  };

  if (!isVisible) {
    return null;
  }

  return (
    <view className={`revenue-details ${isVisible ? "visible" : ""}`}>
      <view className="details-overlay" bindtap={onClose}></view>
      <view className="details-panel">
        {/* 头部 */}
        <view className="details-header">
          <text className="details-title">收益分配详情</text>
          <view className="header-actions">
            <view className="close-button" bindtap={onClose}>
              <Icon name="close" size={20} className="close-icon" />
            </view>
          </view>
        </view>

        <scroll-view
          className="details-content"
          scroll-orientation="vertical"
          scroll-bar-enable={true}
        >
          {/* 基本信息 */}
          <view className="basic-info">
            <view className="info-item">
              <text className="info-label">总收益：</text>
              <text className="info-value total-amount">
                {distribution.currency} {formatAmount(distribution.totalAmount)}
              </text>
            </view>
            <view className="info-item">
              <text className="info-label">计算时间：</text>
              <text className="info-value">
                {formatCalculatedTime(distribution.calculatedAt)}
              </text>
            </view>
            <view className="info-item calculation-basis">
              <text className="info-label">计算依据：</text>
              <text className="info-value basis-text">
                {distribution.calculationBasis}
              </text>
            </view>
          </view>

          {/* 详细分配列表 */}
          <view className="distribution-section">
            <text className="section-title">分配详情</text>
            <view className="distribution-list">
              {distribution.distributions.map((item, index) => (
                <view
                  key={`${item.party}-${index}`}
                  className="distribution-item"
                >
                  <view className="item-header">
                    <view className="party-info">
                      <view
                        className="category-indicator"
                        style={{
                          backgroundColor: getCategoryColor(
                            item.category,
                            index,
                          ),
                        }}
                      ></view>
                      <text className="party-name">{item.party}</text>
                      <text className="party-category">({item.category})</text>
                    </view>
                    <view className="party-amount">
                      <text className="percentage">{item.percentage}%</text>
                      <text className="amount">
                        {distribution.currency} {formatAmount(item.amount)}
                      </text>
                    </view>
                  </view>

                  {/* 分配理由 */}
                  <text className="distribution-reason">{item.reason}</text>

                  {/* 分配比例可视化条 */}
                  <view className="percentage-bar">
                    <view
                      className="percentage-fill"
                      style={{
                        width: `${item.percentage}%`,
                        backgroundColor: getCategoryColor(item.category, index),
                      }}
                    ></view>
                  </view>
                </view>
              ))}
            </view>
          </view>

          {/* 分配汇总 */}
          <view className="distribution-summary">
            <view className="summary-item">
              <text className="summary-label">总计比例：</text>
              <text className="summary-value">
                {distribution.distributions.reduce(
                  (sum, item) => sum + item.percentage,
                  0,
                )}
                %
              </text>
            </view>
            <view className="summary-item">
              <text className="summary-label">总计金额：</text>
              <text className="summary-value">
                {distribution.currency}{" "}
                {formatAmount(
                  distribution.distributions.reduce(
                    (sum, item) => sum + item.amount,
                    0,
                  ),
                )}
              </text>
            </view>
          </view>
        </scroll-view>
      </view>
    </view>
  );
}
