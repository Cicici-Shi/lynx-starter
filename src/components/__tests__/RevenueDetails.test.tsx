// Copyright 2024 The Lynx Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.

import "@testing-library/jest-dom";
import { describe, it, expect, vi, beforeEach } from "vitest";
import { render, getQueriesForElement } from "@lynx-js/react/testing-library";
import { RevenueDetails } from "../RevenueDetails.tsx";
import type { RevenueDistribution } from "../../types/chatbot.ts";

// Mock the Icon component
vi.mock("../Icon.tsx", () => ({
  Icon: ({ name, size, className }: any) => (
    <view data-testid={`icon-${name}`} className={className}>
      {name}
    </view>
  ),
}));

describe("RevenueDetails Component", () => {
  const mockDistribution: RevenueDistribution = {
    recordId: "test-record-1",
    totalAmount: 10000,
    currency: "CNY",
    calculationBasis: "基于观看时长和互动数据的智能分配算法",
    calculatedAt: "2024-01-15T10:30:00Z",
    distributions: [
      {
        party: "主播",
        percentage: 60,
        amount: 6000,
        reason: "主要内容创作者，获得最大份额",
        category: "streamer",
      },
      {
        party: "平台",
        percentage: 25,
        amount: 2500,
        reason: "提供技术支持和流量推广",
        category: "platform",
      },
      {
        party: "合作方",
        percentage: 15,
        amount: 1500,
        reason: "品牌合作和推广支持",
        category: "partner",
      },
    ],
  };

  const mockOnClose = vi.fn();

  beforeEach(() => {
    vi.clearAllMocks();
  });

  describe("Visibility and Basic Rendering", () => {
    it("should not render when isVisible is false", () => {
      render(
        <RevenueDetails
          distribution={mockDistribution}
          isVisible={false}
          onClose={mockOnClose}
        />,
      );

      // When component returns null, there should be no revenue-details element
      const revenueDetails =
        elementTree.root?.querySelector(".revenue-details");
      expect(revenueDetails).toBeNull();
    });

    it("should render when isVisible is true", () => {
      render(
        <RevenueDetails
          distribution={mockDistribution}
          isVisible={true}
          onClose={mockOnClose}
        />,
      );

      const revenueDetails =
        elementTree.root?.querySelector(".revenue-details");
      expect(revenueDetails).toBeTruthy();
    });

    it("should have correct CSS classes when visible", () => {
      render(
        <RevenueDetails
          distribution={mockDistribution}
          isVisible={true}
          onClose={mockOnClose}
        />,
      );

      const revenueDetails = elementTree.root?.querySelector(
        ".revenue-details.visible",
      );
      expect(revenueDetails).toBeTruthy();
    });

    it("should render main structure components", () => {
      render(
        <RevenueDetails
          distribution={mockDistribution}
          isVisible={true}
          onClose={mockOnClose}
        />,
      );

      expect(elementTree.root?.querySelector(".details-overlay")).toBeTruthy();
      expect(elementTree.root?.querySelector(".details-panel")).toBeTruthy();
      expect(elementTree.root?.querySelector(".details-header")).toBeTruthy();
      expect(elementTree.root?.querySelector(".details-content")).toBeTruthy();
    });
  });

  describe("Header Section", () => {
    beforeEach(() => {
      render(
        <RevenueDetails
          distribution={mockDistribution}
          isVisible={true}
          onClose={mockOnClose}
        />,
      );
    });

    it("should render header with title", () => {
      const header = elementTree.root?.querySelector(".details-header");
      expect(header).toBeTruthy();

      const title = elementTree.root?.querySelector(".details-title");
      expect(title).toBeTruthy();
    });

    it("should render close button", () => {
      const closeButton = elementTree.root?.querySelector(".close-button");
      expect(closeButton).toBeTruthy();
    });
  });

  describe("Basic Information Display", () => {
    beforeEach(() => {
      render(
        <RevenueDetails
          distribution={mockDistribution}
          isVisible={true}
          onClose={mockOnClose}
        />,
      );
    });

    it("should display basic information structure", () => {
      const basicInfo = elementTree.root?.querySelector(".basic-info");
      expect(basicInfo).toBeTruthy();

      const infoItems = elementTree.root?.querySelectorAll(".info-item");
      expect(infoItems?.length).toBeGreaterThan(0);
    });

    it("should display total amount with correct styling", () => {
      const totalAmount = elementTree.root?.querySelector(".total-amount");
      expect(totalAmount).toBeTruthy();
    });

    it("should display calculation basis", () => {
      const calculationBasis =
        elementTree.root?.querySelector(".calculation-basis");
      expect(calculationBasis).toBeTruthy();
    });
  });

  describe("Distribution Details", () => {
    beforeEach(() => {
      render(
        <RevenueDetails
          distribution={mockDistribution}
          isVisible={true}
          onClose={mockOnClose}
        />,
      );
    });

    it("should display distribution section", () => {
      const distributionSection = elementTree.root?.querySelector(
        ".distribution-section",
      );
      expect(distributionSection).toBeTruthy();
    });

    it("should display correct number of distribution items", () => {
      const distributionItems =
        elementTree.root?.querySelectorAll(".distribution-item");
      expect(distributionItems?.length).toBe(3);
    });

    it("should display party information for each item", () => {
      const partyInfos = elementTree.root?.querySelectorAll(".party-info");
      expect(partyInfos?.length).toBe(3);

      const partyNames = elementTree.root?.querySelectorAll(".party-name");
      expect(partyNames?.length).toBe(3);

      const partyCategories =
        elementTree.root?.querySelectorAll(".party-category");
      expect(partyCategories?.length).toBe(3);
    });

    it("should display category indicators", () => {
      const categoryIndicators = elementTree.root?.querySelectorAll(
        ".category-indicator",
      );
      expect(categoryIndicators?.length).toBe(3);
    });

    it("should display percentage bars", () => {
      const percentageBars =
        elementTree.root?.querySelectorAll(".percentage-bar");
      expect(percentageBars?.length).toBe(3);

      const percentageFills =
        elementTree.root?.querySelectorAll(".percentage-fill");
      expect(percentageFills?.length).toBe(3);
    });

    it("should display distribution reasons", () => {
      const distributionReasons = elementTree.root?.querySelectorAll(
        ".distribution-reason",
      );
      expect(distributionReasons?.length).toBe(3);
    });
  });

  describe("Summary Section", () => {
    beforeEach(() => {
      render(
        <RevenueDetails
          distribution={mockDistribution}
          isVisible={true}
          onClose={mockOnClose}
        />,
      );
    });

    it("should display distribution summary", () => {
      const distributionSummary = elementTree.root?.querySelector(
        ".distribution-summary",
      );
      expect(distributionSummary).toBeTruthy();
    });

    it("should display summary items", () => {
      const summaryItems = elementTree.root?.querySelectorAll(".summary-item");
      expect(summaryItems?.length).toBe(2);

      const summaryLabels =
        elementTree.root?.querySelectorAll(".summary-label");
      expect(summaryLabels?.length).toBe(2);

      const summaryValues =
        elementTree.root?.querySelectorAll(".summary-value");
      expect(summaryValues?.length).toBe(2);
    });
  });

  describe("Edge Cases", () => {
    it("should handle empty distributions array", () => {
      const emptyDistribution: RevenueDistribution = {
        ...mockDistribution,
        distributions: [],
      };

      render(
        <RevenueDetails
          distribution={emptyDistribution}
          isVisible={true}
          onClose={mockOnClose}
        />,
      );

      const distributionItems =
        elementTree.root?.querySelectorAll(".distribution-item");
      expect(distributionItems?.length).toBe(0);
    });

    it("should handle single distribution item", () => {
      const singleDistribution: RevenueDistribution = {
        ...mockDistribution,
        distributions: [mockDistribution.distributions[0]],
      };

      render(
        <RevenueDetails
          distribution={singleDistribution}
          isVisible={true}
          onClose={mockOnClose}
        />,
      );

      const distributionItems =
        elementTree.root?.querySelectorAll(".distribution-item");
      expect(distributionItems?.length).toBe(1);
    });

    it("should handle unknown category types", () => {
      const unknownCategoryDistribution: RevenueDistribution = {
        ...mockDistribution,
        distributions: [
          {
            party: "未知方",
            percentage: 100,
            amount: 10000,
            reason: "未知类别测试",
            category: "unknown" as any,
          },
        ],
      };

      render(
        <RevenueDetails
          distribution={unknownCategoryDistribution}
          isVisible={true}
          onClose={mockOnClose}
        />,
      );

      const distributionItems =
        elementTree.root?.querySelectorAll(".distribution-item");
      expect(distributionItems?.length).toBe(1);

      const categoryIndicators = elementTree.root?.querySelectorAll(
        ".category-indicator",
      );
      expect(categoryIndicators?.length).toBe(1);
    });
  });

  describe("Component Structure", () => {
    it("should match expected component structure", () => {
      render(
        <RevenueDetails
          distribution={mockDistribution}
          isVisible={true}
          onClose={mockOnClose}
        />,
      );

      expect(elementTree.root).toMatchSnapshot();
    });

    it("should render with correct CSS classes", () => {
      render(
        <RevenueDetails
          distribution={mockDistribution}
          isVisible={true}
          onClose={mockOnClose}
        />,
      );

      // Check main container classes
      expect(
        elementTree.root?.querySelector(".revenue-details.visible"),
      ).toBeTruthy();
      expect(elementTree.root?.querySelector(".details-overlay")).toBeTruthy();
      expect(elementTree.root?.querySelector(".details-panel")).toBeTruthy();

      // Check section classes
      expect(elementTree.root?.querySelector(".basic-info")).toBeTruthy();
      expect(
        elementTree.root?.querySelector(".distribution-section"),
      ).toBeTruthy();
      expect(
        elementTree.root?.querySelector(".distribution-summary"),
      ).toBeTruthy();
    });
  });

  describe("Animation and Styling", () => {
    it("should apply correct CSS classes for animations", () => {
      render(
        <RevenueDetails
          distribution={mockDistribution}
          isVisible={true}
          onClose={mockOnClose}
        />,
      );

      const revenueDetails =
        elementTree.root?.querySelector(".revenue-details");
      expect(revenueDetails).toBeTruthy();
      expect(revenueDetails?.classList.contains("visible")).toBe(true);
    });

    it("should render percentage bars with correct structure", () => {
      render(
        <RevenueDetails
          distribution={mockDistribution}
          isVisible={true}
          onClose={mockOnClose}
        />,
      );

      const percentageBars =
        elementTree.root?.querySelectorAll(".percentage-bar");
      expect(percentageBars?.length).toBe(3);

      const percentageFills =
        elementTree.root?.querySelectorAll(".percentage-fill");
      expect(percentageFills?.length).toBe(3);
    });
  });
});
