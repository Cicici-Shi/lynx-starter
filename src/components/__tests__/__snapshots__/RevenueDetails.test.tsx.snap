// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RevenueDetails Component > Component Structure > should match expected component structure 1`] = `
<page>
  <view
    class="revenue-details visible"
  >
    <view
      class="details-overlay"
    />
    <view
      class="details-panel"
    >
      <view
        class="details-header"
      >
        <text
          class="details-title"
        >
          收益分配详情
        </text>
        <view
          class="header-actions"
        >
          <view
            class="close-button"
          >
            <view
              class="close-icon"
              data-testid="icon-close"
            >
              close
            </view>
          </view>
        </view>
      </view>
      <scroll-view
        class="details-content"
        scroll-bar-enable="true"
        scroll-orientation="vertical"
      >
        <view
          class="basic-info"
        >
          <view
            class="info-item"
          >
            <text
              class="info-label"
            >
              总收益：
            </text>
            <text
              class="info-value total-amount"
            >
              <wrapper>
                CNY
              </wrapper>
               
              <wrapper>
                10,000.00
              </wrapper>
            </text>
          </view>
          <view
            class="info-item"
          >
            <text
              class="info-label"
            >
              计算时间：
            </text>
            <text
              class="info-value"
            >
              2024/01/15 18:30
            </text>
          </view>
          <view
            class="info-item calculation-basis"
          >
            <text
              class="info-label"
            >
              计算依据：
            </text>
            <text
              class="info-value basis-text"
            >
              基于观看时长和互动数据的智能分配算法
            </text>
          </view>
        </view>
        <view
          class="distribution-section"
        >
          <text
            class="section-title"
          >
            分配详情
          </text>
          <view
            class="distribution-list"
          >
            <view
              class="distribution-item"
            >
              <wrapper>
                <view
                  class="item-header"
                >
                  <view
                    class="party-info"
                  >
                    <view
                      class="category-indicator"
                      style="background-color: rgb(16, 185, 129);"
                    />
                    <text
                      class="party-name"
                    >
                      主播
                    </text>
                    <text
                      class="party-category"
                    >
                      (
                      <wrapper>
                        streamer
                      </wrapper>
                      )
                    </text>
                  </view>
                  <view
                    class="party-amount"
                  >
                    <text
                      class="percentage"
                    >
                      <wrapper>
                        60
                      </wrapper>
                      %
                    </text>
                    <text
                      class="amount"
                    >
                      <wrapper>
                        CNY
                      </wrapper>
                       
                      <wrapper>
                        6,000.00
                      </wrapper>
                    </text>
                  </view>
                </view>
                <text
                  class="distribution-reason"
                >
                  主要内容创作者，获得最大份额
                </text>
                <view
                  class="percentage-bar"
                >
                  <view
                    class="percentage-fill"
                    style="width: 60%; background-color: rgb(16, 185, 129);"
                  />
                </view>
              </wrapper>
            </view>
            <view
              class="distribution-item"
            >
              <wrapper>
                <view
                  class="item-header"
                >
                  <view
                    class="party-info"
                  >
                    <view
                      class="category-indicator"
                      style="background-color: rgb(59, 130, 246);"
                    />
                    <text
                      class="party-name"
                    >
                      平台
                    </text>
                    <text
                      class="party-category"
                    >
                      (
                      <wrapper>
                        platform
                      </wrapper>
                      )
                    </text>
                  </view>
                  <view
                    class="party-amount"
                  >
                    <text
                      class="percentage"
                    >
                      <wrapper>
                        25
                      </wrapper>
                      %
                    </text>
                    <text
                      class="amount"
                    >
                      <wrapper>
                        CNY
                      </wrapper>
                       
                      <wrapper>
                        2,500.00
                      </wrapper>
                    </text>
                  </view>
                </view>
                <text
                  class="distribution-reason"
                >
                  提供技术支持和流量推广
                </text>
                <view
                  class="percentage-bar"
                >
                  <view
                    class="percentage-fill"
                    style="width: 25%; background-color: rgb(59, 130, 246);"
                  />
                </view>
              </wrapper>
            </view>
            <view
              class="distribution-item"
            >
              <wrapper>
                <view
                  class="item-header"
                >
                  <view
                    class="party-info"
                  >
                    <view
                      class="category-indicator"
                      style="background-color: rgb(245, 158, 11);"
                    />
                    <text
                      class="party-name"
                    >
                      合作方
                    </text>
                    <text
                      class="party-category"
                    >
                      (
                      <wrapper>
                        partner
                      </wrapper>
                      )
                    </text>
                  </view>
                  <view
                    class="party-amount"
                  >
                    <text
                      class="percentage"
                    >
                      <wrapper>
                        15
                      </wrapper>
                      %
                    </text>
                    <text
                      class="amount"
                    >
                      <wrapper>
                        CNY
                      </wrapper>
                       
                      <wrapper>
                        1,500.00
                      </wrapper>
                    </text>
                  </view>
                </view>
                <text
                  class="distribution-reason"
                >
                  品牌合作和推广支持
                </text>
                <view
                  class="percentage-bar"
                >
                  <view
                    class="percentage-fill"
                    style="width: 15%; background-color: rgb(245, 158, 11);"
                  />
                </view>
              </wrapper>
            </view>
          </view>
        </view>
        <view
          class="distribution-summary"
        >
          <view
            class="summary-item"
          >
            <text
              class="summary-label"
            >
              总计比例：
            </text>
            <text
              class="summary-value"
            >
              <wrapper>
                100
              </wrapper>
              %
            </text>
          </view>
          <view
            class="summary-item"
          >
            <text
              class="summary-label"
            >
              总计金额：
            </text>
            <text
              class="summary-value"
            >
              CNY
               
              10,000.00
            </text>
          </view>
        </view>
      </scroll-view>
    </view>
  </view>
</page>
`;
